internal/scm/azuredevops/azuredevops.go

```diff
@@ -0,0 +1,540 @@
+			if *repo.Project.Name == repository.ProjectName && *repo.Name == repository.Name {
+				filteredRepos = append(filteredRepos, repo)
+				break
+			}
+		}
+	}
+
+	return &filteredRepos, nil
+}
+
+func (a *AzureDevOps) CreatePullRequest(ctx context.Context, _ scm.Repository, prRepo scm.Repository, newPR scm.NewPullRequest) (scm.PullRequest, error) {
+	prr := prRepo.(repository)
+
+	reviewerIDs, err := a.getUserIds(ctx, append(newPR.Reviewers, newPR.Assignees...))
+	if err != nil {
+		return nil, err
+	}
+
+	reviewers := make([]git.IdentityRefWithVote, len(reviewerIDs))
+	for i, reviewerID := range reviewerIDs {
+		rCopy := reviewerID
+		reviewers[i] = git.IdentityRefWithVote{
+			Id: &rCopy,
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	prTitle := newPR.Title
+	if newPR.Draft {
+		prTitle = "Draft: " + prTitle
+	}
+
+	activeLabel := true
+	labels := make([]core.WebApiTagDefinition, len(newPR.Labels))
+	removeSourceBranch := true
+
+	for i, label := range newPR.Labels {
+		lCopy := label
+		labels[i] = core.WebApiTagDefinition{
+			Active: &activeLabel,
+			Name:   &lCopy,
+		}
+	}
+	srn := PrependPrefixIfNeeded(newPR.Head) // Pass the value of newPR.Head
+	trn := PrependPrefixIfNeeded(newPR.Base)
+
+	requestArgs := git.CreatePullRequestArgs{
+		GitPullRequestToCreate: &git.GitPullRequest{
+			Title:         &prTitle,
+			Description:   &newPR.Body,
+			SourceRefName: &srn,
+			TargetRefName: &trn,
+			CompletionOptions: &git.GitPullRequestCompletionOptions{
+				DeleteSourceBranch: &removeSourceBranch,
+			},
+			Labels:    &labels,
+			Reviewers: &reviewers,
+			IsDraft:   &newPR.Draft,
+		},
+		RepositoryId: &prr.rid,
+		Project:      &prr.ownerName,
+	}
+
+	pr, err := a.gitClient.CreatePullRequest(ctx, requestArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	return pullRequest{
+		ownerName:  prr.ownerName,
+		repoName:   prr.name,
+		repoID:     prr.rid,
+		branchName: newPR.Head,
+		id:         *pr.PullRequestId,
+	}, nil
+}
+
+func (a *AzureDevOps) UpdatePullRequest(ctx context.Context, repo scm.Repository, pullReq scm.PullRequest, updatedPR scm.NewPullRequest) (scm.PullRequest, error) {
+	r := repo.(repository)
+	pr := pullReq.(pullRequest)
+
+	reviewerIDs, err := a.getUserIds(ctx, append(updatedPR.Reviewers, updatedPR.Assignees...))
+	if err != nil {
+		return nil, err
+	}
+
+	reviewers := make([]git.IdentityRefWithVote, len(reviewerIDs))
+	for i, reviewerID := range reviewerIDs {
+		rCopy := reviewerID
+		reviewers[i] = git.IdentityRefWithVote{
+			Id: &rCopy,
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	prTitle := updatedPR.Title
+	if updatedPR.Draft {
+		prTitle = "Draft: " + prTitle
+	}
+
+	activeLabel := true
+	labels := make([]core.WebApiTagDefinition, len(updatedPR.Labels))
+
+	for i, label := range updatedPR.Labels {
+		lCopy := label
+		labels[i] = core.WebApiTagDefinition{
+			Active: &activeLabel,
+			Name:   &lCopy,
+		}
+	}
+
+	srn := PrependPrefixIfNeeded(updatedPR.Head) // Pass the value of newPR.Head
```

Comment:
I'm not sure I understand the comment here.

Reply:
removed that